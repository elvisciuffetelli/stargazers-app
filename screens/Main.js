import React from "react";
import { ScrollView } from "react-native";

import Search from "../components/Search";
import Heading from "../components/Heading";
import List from "../components/List";
import useFetchUsers from "../hooks/useFetchUsers";
import styles from "../styles";

export default function Main() {
  const {
    isLoading,
    users,
    fieldError,
    errorMessage,
    handlePress,
    onChangeOwner,
    onChangeRepo,
    owner,
    repo,
  } = useFetchUsers();

  return (
    <ScrollView style={styles.container}>
      <Heading
        title="Stargazers app"
        subtitle="Find out a repository stargazers"
      />
      <Search
        handlePress={handlePress}
        onChangeOwner={onChangeOwner}
        owner={owner}
        onChangeRepo={onChangeRepo}
        repo={repo}
        fieldError={fieldError}
        errorMessage={errorMessage}
      />
      <List
        isLoading={isLoading}
        users={users}
      />
    </ScrollView>
  );
}
