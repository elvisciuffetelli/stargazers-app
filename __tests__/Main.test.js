import React from "react";
import { render, screen } from '@testing-library/react-native';

import Main from "../screens/Main";

describe("<Main />", () => {
  test('It must render and match the snapshot', async () => {
    render(<Main />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
