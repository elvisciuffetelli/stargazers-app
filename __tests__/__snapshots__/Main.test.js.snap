// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> It must render and match the snapshot 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#f3f3f3",
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Object {
            "color": "#242424",
            "fontSize": 51.42857142857143,
            "fontWeight": "300",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Stargazers app
      </Text>
      <Text
        accessibilityRole="text"
        style={
          Object {
            "color": "#242424",
            "fontSize": 36,
            "fontWeight": "300",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Find out a repository stargazers
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
        testID="RNE__Input__view-wrapper"
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            editable={true}
            onChangeText={[Function]}
            placeholder="Repo owner"
            placeholderTextColor="#86939e"
            style={
              Object {
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "firebrick",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
        testID="RNE__Input__view-wrapper"
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            editable={true}
            onChangeText={[Function]}
            placeholder="Repo name"
            placeholderTextColor="#86939e"
            style={
              Object {
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "firebrick",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "#2089dc",
                "borderRadius": 30,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 20,
                "padding": 8,
                "paddingHorizontal": 10,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              SEARCH
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 20,
        }
      }
    />
  </View>
</RCTScrollView>
`;
